File: app/src/main/java/com/satanasov/phonebook/presenter/ContactActivityPresenter.java
package com.satanasov.phonebook.presenter;
import android.content.Context;
import com.satanasov.phonebook.db.DataBaseQueries;
import com.satanasov.phonebook.model.ContactModel;

public class ContactActivityPresenter extends BasePresenter<ContactActivityView> {

    private DataBaseQueries     mDataBaseQueries  = new DataBaseQueries();
    private ContactActivityView mContactActivityView;
    public boolean              mIsContactForEdit = false;


   public ContactActivityPresenter(ContactActivityView contactActivityView){
       attachView(contactActivityView);
       this.mContactActivityView = contactActivityView;
   }

    @Override
    public void subscribe() {
        super.subscribe();
        if (mIsContactForEdit)
            view.setContactDetailsForEdit();
        else
            view.setContactDetailsForInsert();

    }

    public void saveContactToDB(ContactModel contactModel){

       //mDataBaseQueries.storeContact((Context)mContactActivityView,contactModel);
       view.returnToMainActivity();
    }

    public void updateContactInDB(ContactModel contactModel){
       //mDataBaseQueries.updateContact((Context)mContactActivityView,contactModel);
       view.returnToMainActivity();
    }
}

File: app/src/main/java/com/satanasov/phonebook/kotlinPhoneBook/presenter/ContactActivityPresenter.kt
package com.satanasov.phonebook.kotlinPhoneBook.presenter
import android.content.Context
import com.satanasov.phonebook.db.DataBaseQueries
import com.satanasov.phonebook.kotlinPhoneBook.model.ContactModel

class ContactActivityPresenter(contactActivityView: ContactActivityView) : BasePresenter<ContactActivityView>() {
    private var mDataBaseQueries: DataBaseQueries = DataBaseQueries()
    private var mContactActivityView: ContactActivityView
    var mIsContactForEdit: Boolean = false

    init{
        attachView(contactActivityView)
        this.mContactActivityView = contactActivityView
    }

    override fun subscribe() {
        super.subscribe()
        if (mIsContactForEdit)
            view!!.setContactDetailsForEdit()
        else
            view!!.setContactDetailsForInsert()
    }

    fun saveContactToDB(contactModel: ContactModel){
        mDataBaseQueries.storeContact(mContactActivityView as Context,contactModel)
        view?.returnToMainActivity()
    }

    fun updateContactInDb(contactModel: ContactModel){
        mDataBaseQueries.updateContact(mContactActivityView as Context,contactModel)
        view?.returnToMainActivity()
    }
}